static/src/javascripts/projects/commercial/modules/messenger/background.spec.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+import type { JestMockT } from 'jest';
+
+import { register } from 'commercial/modules/messenger';
+import { _ } from './background';
+
+const { setBackground, getStylesFromSpec } = _;
+
+// Jest understands `register.mock.calls`, however, Flow gets angry because:
+// 'property mock not found in statics of function'. This is a helper to allow
+// `expect(register.mock.calls[0][0]).toBe('background');` to pass linting
+const foolFlow = (mockFn: any) => ((mockFn: any): JestMockT);
+
+jest.mock('commercial/modules/messenger', () => ({
+    register: jest.fn(),
+}));
+
+const adSpec = {
+    scrollType: 'fixed',
+    backgroundColour: 'ffffff',
+    backgroundImage: 'image',
+    backgroundRepeat: 'no-repeat',
+    backgroundPosition: 'absolute',
+};
+
+describe('Cross-frame messenger: setBackground', () => {
+    beforeEach(done => {
+        if (document.body) {
+            document.body.innerHTML = `
+              <div>
+                  <div id="slot01"><div id="iframe01" class="iframe"></div></div>
+              </div>`;
+        }
+        done();
+    });
+
+    it('should call register', () => {
+        expect(register).toHaveBeenCalled();
+        expect(foolFlow(register).mock.calls[0][0]).toBe('background');
+    });
+
+    it('should return nothing if there are no specs', () => {
+        const fakeAdSlot = document.createElement('div');
+        // Ignore the missing argument, becuase we want to test that... $FlowFixMe
+        expect(setBackground({}, fakeAdSlot)).toBeNull();
+    });
+
+    it('should create new elements if there are specs', done => {
+        const fallback = document.createElement('div');
+        const fakeAdSlot = document.getElementById('slot01') || fallback;
+
+        foolFlow(setBackground(adSpec, fakeAdSlot))
```

Comment:
`setBackground` can potentially return `null` (it won't in this test scenario, but Flow gets unhappy at the possibility)

I'm using foolFlow to remove the typing and subsequent errors from the test:
![picture 799](https://user-images.githubusercontent.com/8607683/27581862-d7e489fe-5b26-11e7-809d-ef07d22c611f.png)

Using $FlowFixMe to silence them also works; what is your preference?